### PR TITLE
Add `PaywallFooterView` listeners

### PR DESCRIPTION
**Pull request by tonidero**
This PR adds initial listeners to the `PaywallFooterView`. Follow up from #995

### CODE REVIEW

File: purchases_ui_flutter/ios/Classes/PurchasesUiPaywallFooterView.swift [hunk 1]
```diff
@@ -29,51 +34,76 @@ class PurchasesUiPaywallFooterViewFactory: NSObject, FlutterPlatformViewFactory
...
+        _paywallProxy?.delegate = self;
     }
 
     func view() -> UIView {
         return _view
     }
+}
 
-    @available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
-    func paywallViewController(_ controller: PaywallViewController, didChangeSizeTo size: CGSize) {
```
Comment (tonidero) [inline purchases_ui_flutter/ios/Classes/PurchasesUiPaywallFooterView.swift:74]: I hadn't actually setup the delegate in #999 so this was never called. After actually setting the delegate, this was returning a different value than the size as given by the view initially, which was making it draw incorrectly. I need to revisit whether this is needed and if so, why is it returning the wrong size...

File: purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallFooterView.kt [hunk 2]
```diff
@@ -32,7 +33,7 @@ internal class PaywallFooterView(
     override fun dispose() {}
 
     init {
-        methodChannel = MethodChannel(messenger, "purchases_ui_flutter/PaywallFooterView/${id}")
+        methodChannel = MethodChannel(messenger, "com.revenuecat.purchasesui/PaywallFooterView/${id}")
```
Comment (tonidero) [inline purchases_ui_flutter/android/src/main/kotlin/com/revenuecat/purchases_ui_flutter/views/PaywallFooterView.kt:36]: Changed the channel name to be consistent with the full screen version

File: purchases_ui_flutter/lib/views/paywall_view.dart [hunk 3]
```diff
@@ -22,6 +24,16 @@ import 'package:purchases_flutter/models/store_transaction.dart';
...
+///
+/// [onPurchaseError] (Optional) Callback that gets called when a purchase
+/// fails.
+///
+/// [onRestoreCompleted] (Optional) Callback that gets called when a restore
+/// is completed. Note that this may get called even if no entitlements have
+/// been granted in case no relevant purchases were found.
+///
+/// [onRestoreError] (Optional) Callback that gets called when a restore
+/// fails.
```
Comment (vegaro) [inline purchases_ui_flutter/lib/views/paywall_view.dart:36]: thanks for adding these!

File: purchases_ui_flutter/lib/views/paywall_view_method_handler.dart [hunk 4]
```diff
@@ -0,0 +1,80 @@
+
+import 'package:flutter/services.dart';
+import 'package:purchases_flutter/models/customer_info_wrapper.dart';
+import 'package:purchases_flutter/models/package_wrapper.dart';
+import 'package:purchases_flutter/models/purchases_error.dart';
+import 'package:purchases_flutter/models/store_transaction.dart';
+
+class PaywallViewMethodHandler {
```
Comment (vegaro) [inline purchases_ui_flutter/lib/views/paywall_view_method_handler.dart:8]: so clean!

**Review by vegaro — APPROVED** [review]
Thanks for all that cleanup!

**Pull request merged**